src/views/settings/view_protection_settings.cpp

```diff
@@ -50,6 +50,29 @@ namespace big
 			ImGui::SetTooltip("This prevents any pickup from the ground such as unwanted money drops.\nAttention: Normal pickups are also no longer possible.");
 		ImGui::Checkbox("Admin Check", &g.protections.admin_check);
 		ImGui::EndGroup();
+
+		ImGui::BeginGroup();
+		components::sub_title("Blocked IPs");
+		for (auto& element : g.protections.ip_block_list)
+		{
+			char* text = const_cast<char*>(element.c_str());
+			ImGui::Text(text);
+			if (ImGui::Button("Remove"))
```

Comment:
Instead of making two separate UI elements, combine them in a single one. Preferably one that employs some sort of label in which the IP could be displayed and its operation connected to removal. Such as a ImGui::Selectable with the flag ImGuiSelectableFlags_AllowDoubleClick (This method would benefit from a listbox). The button itself could even contain the blocked Ip (with some imgui magic to make it look half decent).

Reply:
I sense element id complications here as well. PushId and PopId would suffice.

Reply:
So I should make a new function or is that a function of the vector?

Reply:
Iirc anything other than ImGui::Text, requires an unique id (the labels). Otherwise the UI elements won't behave as expected. In such scopes as loops, creating an UI element is like creating multiple. In which case you either handle the Id's yourself using the ## identifier or let the library handle it by using ImGui::PushId and ImGui::PopId respectively. Besides the id problem, the method you have employed won't be pleasing to the eye. Id advise you look at similar features in the code base and adapt.

Reply:
So I add an ImGui::PushId before calling the button in the loop?

Reply:
Yes. Keep in mind it requires an unique identifier param. You could use the ip itself.